### PR TITLE
feat: adding support for gcp impersonated credentials

### PR DESCRIPTION
**Pull request by TigerC10**
If a service account email address is found in the `GOOGLE_IMPERSONATE_SERVICE_ACCOUNT` environment variable, gcp client will attempt to create an oauth token impersonating that service account to use as credentials.

closes https://github.com/cloud-custodian/cloud-custodian/issues/8568

### CODE REVIEW

**Comment by kapilt** [comment]
thanks for the pr, we'll want to update the docs as well re usage, probably a note in gcp's getting started on environment variables

**Comment by TigerC10** [comment]
@kapilt ask and ye shall receive.

File: tools/c7n_gcp/c7n_gcp/client.py [hunk 1]
```diff
@@ -177,9 +179,20 @@ def __init__(self,
...
+            default_credentials, _ = google.auth.default(quota_project_id=project_id or
             get_default_project())
-        self._credentials = with_scopes_if_required(credentials, list(CLOUD_SCOPES))
+        if GOOGLE_IMPERSONATE_SERVICE_ACCOUNT:
+            impersonated_credentials = google.auth.impersonated_credentials.Credentials(
+                source_credentials=credentials or default_credentials,
+                target_principal=GOOGLE_IMPERSONATE_SERVICE_ACCOUNT,
+                target_scopes=list(CLOUD_SCOPES))
+        target_credentials = impersonated_credentials or credentials or default_credentials
+        if not impersonated_credentials:
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/client.py:190]: if the environment variable isn't present this fails as an undefined variable.

Comment (TigerC10) [reply tools/c7n_gcp/c7n_gcp/client.py]: Great catch, I came at this like a javascript developer so I've updated it to something more appropriate for Python.

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Comment by kapilt** [comment]
I went ahead and added support for passing service account via cli flag --assume service_account_email, update docs for that, and fixed the undefined var issue. cli flag takes precedence over env var.

**Pull request merged**